addons/web/static/src/views/helpers/view_hook.js

```diff
@@ -36,3 +39,79 @@ export function useViewArch(arch, params = {}) {
 
     return processedArch;
 }
+
+/**
+ * Allows for a topmost component exclusively (usually a View component)
+ * to handle links with attribute type="action". This is used to support onboarding banners
+ * and content helpers.
+ *
+ * Note that this is similar but quite different from action buttons, since action links
+ * are not dynamic according to the record.
+ * @param {Object} params
+ * @param  {String} params.resModel The default resModel to which actions will apply
+ * @param  {KeepLast} [params.keepLast] The view's keeplast to allow concurrency management.
+ *   A new one is created if none is passed
+ */
+export function useActionLinks({ keepLast, resModel }) {
+    const selector = `a[type="action"]`;
+    const component = owl.hooks.useComponent();
+    const orm = useService("orm");
+    const { doAction } = useService("action");
+    keepLast = keepLast || new KeepLast();
+
+    async function handler(ev) {
+        ev.preventDefault();
+        const target = ev.target;
+        const data = target.dataset;
+
+        if (data.method !== undefined && data.model !== undefined) {
+            const options = {};
+            if (data.reloadOnClose) {
+                options.onClose = () => component.render();
+            }
+            const action = await keepLast.add(orm.call(data.model, data.method));
+            if (action !== undefined) {
+                keepLast.add(doAction(action, options));
+            }
+        } else if (target.getAttribute("name")) {
+            const options = {};
+            if (data.context) {
+                options.additionalContext = evaluateExpr(data.context);
+            }
+            keepLast.add(doAction(target.getAttribute("name"), options));
+        } else {
+            let views;
+            const resId = data.resid ? parseInt(data.resid, 10) : null;
+            if (data.views) {
+                views = evaluateExpr(data.views);
+            } else {
+                views = resId
+                    ? [[false, "form"]]
+                    : [
+                          [false, "list"],
+                          [false, "form"],
+                      ];
+            }
+
+            const action = {
+                name: target.getAttribute("title") || target.textContent.trim(),
+                type: "ir.actions.act_window",
+                res_model: data.model || resModel,
+                target: "current", // TODO: make customisable?
```

Comment:
didn't you mean "main"?

Reply:
I copied the original code from AbstractController._onActionClicked ...